minions/async/src/main/java/io/github/microcks/minion/async/producer/BindingProducerManager.java

```diff
@@ -0,0 +1,110 @@
+package io.github.microcks.minion.async.producer;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.jboss.logging.Logger;
+
+import io.github.microcks.domain.EventMessage;
+import io.github.microcks.minion.async.AsyncMockDefinition;
+import io.github.microcks.util.URIBuilder;
+import io.github.microcks.util.el.TemplateEngine;
+import io.github.microcks.util.el.TemplateEngineFactory;
+
+public abstract class BindingProducerManager {
```

Comment:
I  introduced this base class to remove duplicate code (topic string creation) and in order to simplify the ProducerManager class (the case statement switching between bindings).

There's a template method publishMessages that makes sure topic strings are created and messages are sent.
Sub-classes can extend the topic generation (see the KafkaProducer) and need to implement a publishOne() method.